test/feeder_ex_test.exs

```diff
@@ -30,6 +30,11 @@ defmodule FeederExTest do
 
   end
 
+  test "parse! returns just the parsed results on success" do
+    parsed_feed = FeederEx.parse!(File.read!(@sample_file))
+    assert parsed_feed.title == "Liftoff News"
+  end
```

Comment:
So, I did do a test locally with a badly formed feed (just `"<xml>"` and nothing else). It raised the following:

```

Interactive Elixir (1.3.2) - press Ctrl+C to exit (type h() ENTER for help)
pry(1)> FeederEx.parse!("<xml>")
** (BadMapError) expected a map, got: nil
    (feeder_ex) lib/feeder_ex/parser.ex:39: FeederEx.Parser.event/2
                xmerl_sax_parser_utf8.erl:3309: :xmerl_sax_parser_utf8.event_callback/2
                xmerl_sax_parser_utf8.erl:121: :xmerl_sax_parser_utf8.parse/2
    (feeder_ex) lib/feeder_ex.ex:26: FeederEx.parse!/1
       (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
          (iex) lib/iex/evaluator.ex:135: IEx.Evaluator.handle_eval/6
          (iex) lib/iex/evaluator.ex:128: IEx.Evaluator.do_eval/4
          (iex) lib/iex/evaluator.ex:108: IEx.Evaluator.eval/4
          (iex) lib/iex/evaluator.ex:27: IEx.Evaluator.loop/3
          (iex) lib/iex/evaluator.ex:18: IEx.Evaluator.init/3
          (iex) lib/iex.ex:439: IEx.pry/3
                test/feeder_ex_test.exs:35: FeederExTest."test parse! returns just the parsed results on success"/1
      (ex_unit) lib/ex_unit/runner.ex:296: ExUnit.Runner.exec_test/1
       (stdlib) timer.erl:166: :timer.tc/1
      (ex_unit) lib/ex_unit/runner.ex:246: anonymous fn/3 in ExUnit.Runner.spawn_test/3
```

So, my question is, what would you suggest to feed to the example case for failing. Perhaps leaving it as is and allowing the error to propagate through is actually the right thing?